src/Security.php

```diff
@@ -42,7 +42,16 @@ public static function site_url($url, $path, $scheme) {
    */
   public static function wp_redirect($url) {
     $custom_path = defined('CORE_STANDARDS_LOGIN_PATH') ? CORE_STANDARDS_LOGIN_PATH : '/login.php';
-    $url = strtr($url, ['/wp-login.php' => $custom_path]);
+
+    // wp-login.php calls wp_safe_redirect() with a relative path, which causes
+    // wp_validate_redirect() to automatically prepend the current folder name
+    // of the request URI to it. Prevent the path from being duplicated.
+    if (strpos($url, dirname($custom_path)) === FALSE) {
+      $url = strtr($url, ['wp-login.php' => $custom_path]);
```

Comment:
I removed the leading slash as `wp-login.php` not always add one to the URI https://github.com/WordPress/WordPress/blob/master/wp-login.php#L808